### PR TITLE
Relogin

### PR DESCRIPTION
**Pull request by tomaszwarda**

### CODE REVIEW

**Review by danielui — APPROVED** [review]

File: vmngclient/session.py [hunk 1]
```diff
@@ -168,6 +172,8 @@ def __init__(
         self._session_type = SessionType.NOT_DEFINED
         self.server_name = None
         self.logger = logging.getLogger(__name__)
+        self.enable_relogin: bool = True
+        self.__second_relogin_try: bool = False
```
Comment (tehAgitto) [inline vmngclient/session.py:176]: The flag should be replaced by some counter which will say how many times we should try to login

File: vmngclient/session.py [hunk 2]
```diff
@@ -189,6 +195,17 @@ def request(self, method, url, *args, **kwargs) -> vManageResponse:
...
+            if not self.__second_relogin_try:
+                self.logger.warning("Relogin session...")
+                self.auth = vManageAuth(self.base_url, self.username, self.password, verify=False)
+                self.__second_relogin_try = True
+                return self.request(method, url, *args, **kwargs)
+            else:
+                self.logger.warning("Relogin failed.")
+        else:
+            self.__second_relogin_try = False
+
```
Comment (tehAgitto) [inline vmngclient/session.py:208]: Shouldn't we keep connection through an `auth` object instead of creating new one?

**Review by tehAgitto — CHANGES_REQUESTED** [review]

File: vmngclient/session.py (hunk 1 above)
Comment (sbasan) [reply vmngclient/session.py:176]: @tehAgitto AFAIK this flag was introduced to disable re-login on purpose (negative tests etc.. By default it should be enabled and advanced used can disable it after session instance is created.

Comment (sbasan) [reply vmngclient/session.py:176]: @tehAgitto Ok now i get it you want user to set max_relogin_attempts, and disabling re-login with value 0.

Comment (sbasan) [reply vmngclient/session.py:177]: @tehAgitto during discussion we found it makes no sense to do more relogin attempts in case when cookie expired.

**Review by sbasan — DISMISSED** [review]
lgtm

**Review by tzietkowski — APPROVED** [review]

**Review by sbasan — DISMISSED** [review]
lgtm

**Review by tehAgitto — CHANGES_REQUESTED** [review]

**Review by tehAgitto — APPROVED** [review]

**Review by sbasan — APPROVED** [review]
lgtm

**Pull request merged**